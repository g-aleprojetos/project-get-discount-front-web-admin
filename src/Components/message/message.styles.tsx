import styled from 'styled-components';
import {TextH1, TextH2} from 'Components/text';

export interface Props {
  colorTitle?: string;
  colorMessege?: string;
}

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 400px;
`;

export const ContainerTexto = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  margin-bottom: 5px;
`;

export const TextTitle = styled(TextH1).attrs({
  peso: 'bold',
})<Props>`
  color: ${props => props.colorTitle};
`;
export const TextMessege = styled(TextH2)<Props>`
  color: ${props => props.colorMessege};
  text-align: center;
`;
