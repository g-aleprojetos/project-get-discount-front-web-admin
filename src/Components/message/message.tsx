import React from 'react';
import * as S from './message.styles';

type PropsMessage = S.Props & {
  testId?: string;
  textTitle?: string;
  textMessage?: string;
};

export const Message = (props: PropsMessage) => {
  const {
    testId,
    textTitle = '',
    textMessage = '',
    colorTitle,
    colorMessege,
  } = props;
  return (
    <S.Container data-testid={testId ?? 'message-pardao'}>
      <S.ContainerTexto>
        <S.TextTitle colorTitle={colorTitle}>{textTitle}</S.TextTitle>
      </S.ContainerTexto>
      <S.ContainerTexto>
        <S.TextMessege colorMessege={colorMessege}>{textMessage}</S.TextMessege>
      </S.ContainerTexto>
    </S.Container>
  );
};
