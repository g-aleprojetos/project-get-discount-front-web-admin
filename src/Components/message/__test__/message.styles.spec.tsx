import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../message.styles';

describe('message.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerTexto DEVE ser igual ao snapshot', () => {
    const containerTexto = renderer.create(<S.ContainerTexto />).toJSON();

    expect(containerTexto).toMatchSnapshot();
  });

  test('TextTitle DEVE ser igual ao snapshot', () => {
    const textTitle = renderer.create(<S.TextTitle />).toJSON();

    expect(textTitle).toMatchSnapshot();
  });

  test('TextTitle DEVE ser igual ao snapshot', () => {
    const textTitle = renderer
      .create(<S.TextTitle colorTitle="#3D9DB3" />)
      .toJSON();

    expect(textTitle).toMatchSnapshot();
  });

  test('TextMessege DEVE ser igual ao snapshot', () => {
    const textMessege = renderer.create(<S.TextMessege />).toJSON();

    expect(textMessege).toMatchSnapshot();
  });

  test('TextMessege DEVE ser igual ao snapshot', () => {
    const textMessege = renderer
      .create(<S.TextMessege colorMessege="#3D9DB3" />)
      .toJSON();

    expect(textMessege).toMatchSnapshot();
  });
});
