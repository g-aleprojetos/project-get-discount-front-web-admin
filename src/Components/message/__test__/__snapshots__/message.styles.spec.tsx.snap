// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 400px;
}

<div
  className="c0"
/>
`;

exports[`message.styles ContainerTexto DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-bottom: 5px;
}

<div
  className="c0"
/>
`;

exports[`message.styles TextMessege DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 26px;
  cursor: text;
}

.c1 {
  text-align: center;
}

<h2
  className="c0 c1"
/>
`;

exports[`message.styles TextMessege DEVE ser igual ao snapshot 2`] = `
.c0 {
  font-size: 18px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 26px;
  cursor: text;
}

.c1 {
  color: #3D9DB3;
  text-align: center;
}

<h2
  className="c0 c1"
/>
`;

exports[`message.styles TextTitle DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Bold;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 34px;
  cursor: text;
}

<h1
  className="c0 "
/>
`;

exports[`message.styles TextTitle DEVE ser igual ao snapshot 2`] = `
.c0 {
  font-size: 24px;
  font-family: AvertaStd-Bold;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 34px;
  cursor: text;
}

.c1 {
  color: #3D9DB3;
}

<h1
  className="c0 c1"
/>
`;
